src/soc/ibm/power9/homer.c

```diff
@@ -997,6 +1076,244 @@ static void istep_16_1(int this_core)
+
+	config_data->version = OCC_HOST_DATA_VERSION;
+	config_data->nest_freq = powerbus_cfg()->fabric_freq;
+	config_data->interrupt_type = USE_PSIHB_COMPLEX;
+	config_data->is_fir_master = false;
+	config_data->is_smf_mode = false;
+}
+
+static void load_pm_complex(struct homer_st *homer)
+{
+	/* Common OCC area is located right after HOMER image */
+	uint8_t *common_area = (uint8_t *)homer + sizeof(*homer);
+
+	// TODO resetPMComplex(); also used in istep_6_11.c
+	load_occ_setup(homer, common_area);
+	load_occ_image_to_homer(homer);
+	load_host_data_to_homer(homer);
+}
+
+/* Generates host configuration vector and updates the value in HOMER */
+static void check_proc_config(struct homer_st *homer)
+{
+	uint64_t vector_value = INIT_CONFIG_VALUE;
+	/* XXX: how come this points into padding area? */
+	uint64_t *conf_vector = (void *)((uint8_t *)&homer->qpmr + QPMR_PROC_CONFIG_POS);
+
+	// XXX: where to find these?
+	/* checkChiplet<fapi2::TARGET_TYPE_MCS >( i_procTgt, fapi2::TARGET_TYPE_MCS, vector_value, MCS_POS ); */
+	/* checkChiplet<fapi2::TARGET_TYPE_XBUS>( i_procTgt, fapi2::TARGET_TYPE_XBUS, vector_value, XBUS_POS ); */
+	/* checkChiplet<fapi2::TARGET_TYPE_PHB>( i_procTgt, fapi2::TARGET_TYPE_PHB, vector_value, PHB_POS ); */
+	/* checkChiplet<fapi2::TARGET_TYPE_CAPP>( i_procTgt, fapi2::TARGET_TYPE_CAPP, vector_value, CAPP_POS ); */
+
+	/* checkObusChipletHierarchy(i_procTgt, vector_value, OBUS_POS, NVLINK_POS); */
+
+	/* checkChiplet<fapi2::TARGET_TYPE_MCA>(i_procTgt, fapi2::TARGET_TYPE_MCA, vector_value, MBA_POS); */
+
+	*conf_vector = htobe64(vector_value);
+}
+
+static void special_wakeup_disable(uint64_t cores)
+{
+	enum { PPM_SPWKUP_FSP = 0x200F010B };
+
+	for (int i = 0; i < MAX_CORES_PER_CHIP; ++i) {
+		uint32_t spwkup_addr = PPM_SPWKUP_FSP + 0x01000000 * i;
+
+		if (!IS_EC_FUNCTIONAL(i, cores))
+			continue;
+
+		write_scom(spwkup_addr, 0);
+		(void)read_scom(spwkup_addr);
+	}
+}
+
+static uint32_t ppc_lis(uint16_t rt, uint16_t data)
+{
+	uint32_t inst;
+	inst = LIS_OP;
+	inst |= rt << (31 - 10);
+	inst |= data;
+	return inst;
+}
+
+static uint32_t ppc_ori(uint16_t rs, uint16_t ra, uint16_t data)
+{
+	uint32_t inst;
+	inst = ORI_OP;
+	inst |= rs << (31 - 10);
+	inst |= ra << (31 - 15);
+	inst |= data;
+	return inst;
+}
+
+static uint32_t ppc_mtspr(uint16_t rs, uint16_t spr)
+{
+	enum { MTSPR_CONST1 = 467 };
+
+	uint32_t temp = ((spr & 0x03FF) << (31 - 20));
+
+	uint32_t inst;
+	inst = MTSPR_OP;
+	inst |= rs << (31 - 10);
+	inst |= (temp & 0x0000F800) << 5;  // Perform swizzle
+	inst |= (temp & 0x001F0000) >> 5;  // Perform swizzle
+	inst |= MTSPR_CONST1 << 1;
+	return inst;
+}
+
+static uint32_t ppc_bctr(void)
+{
+	enum { BCCTR_CONST1 = 528 };
+
+	uint32_t inst;
+	inst = BCCTR_OP;
+	inst |= 20 << (31 - 10); // BO
+	/* BI = 0 is taken care of by inst = 0 */
+	inst |= BCCTR_CONST1 << 1;
+	return inst;
+}
+
+static uint32_t ppc_b(uint32_t target_addr)
+{
+	uint32_t inst;
+	inst = BR_OP << (31 - 5);
```

Comment:
BR_OP is already shifted, so:
```suggestion
	inst = BR_OP;
```
No idea why it works in this form...

Reply:
This is from an old commit.